include/aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp

```diff
@@ -8,67 +8,74 @@
+    ::dart::dynamics::ChainPtr finger, size_t proximal, size_t distal,
+    ::dart::collision::CollisionDetectorPtr collisionDetector = nullptr,
+    ::dart::collision::CollisionGroupPtr collideWith = nullptr,
+    ::dart::collision::CollisionOption collisionOptions
       = ::dart::collision::CollisionOption(false, 1));
 
   /// Sets variables to move the finger joints.
-  /// proximal dof moves to _goalPosition, joint limit, or until collision.
+  /// proximal dof moves to goalPosition, joint limit, or until collision.
   /// Step method should be called multiple times until future returns.
-  /// Distal dof follows with mimic ratio. 
-  /// \param _goalPosition Desired angle of proximal joint.
-  /// \param _collideWith CollisionGroup to check collision with fingers.
+  /// Distal dof follows with mimic ratio.
+  /// \param goalPosition Desired angle of proximal joint.
   /// \return future Becomes available when the execution completes.
-  std::future<void> execute(double _goalPosition,
-    ::dart::collision::CollisionGroupPtr _collideWith);
+  std::future<void> execute(double goalPosition);
 
-  /// Returns mimic ratio, i.e. how much the distal joint moves relative to 
+  /// Returns mimic ratio, i.e. how much the distal joint moves relative to
   /// the proximal joint. The joint movements follow
   /// this ratio only when both joints are moving.
   /// \return mimic ratio.
-  static double getMimicRatio();
+  constexpr static double getMimicRatio() { return kMimicRatio; }
 
-  /// Moves the joints of the finger by dofVelocity*_timeSIncePreviousCall
+  /// Moves the joints of the finger by dofVelocity*timeSincePreviousCall
   /// until execute's goalPosition by primary dof or collision is detected.
-  /// If proximal link is in collision, distal link moves until 
+  /// If proximal link is in collision, distal link moves until
   /// mimicRatio*goalPosition. If distal link is in collision, execution stops.
   /// If multiple threads are accessing this function or skeleton associated
   /// with this executor, it is necessary to lock the skeleton before
   /// calling this method.
-  /// \param _timeSincePreviousCall Time since previous call. 
-  void step(double _timeSincePreviousCall);
+  /// \param[in] timeSincePreviousCall Time interval to take.
+  void step(const std::chrono::milliseconds& timeSincePreviousCall);
+
+  /// Resets CollisionGroup to check collision with fingers.
+  /// \param _collideWith CollisionGroup to check collision with fingers.
+  /// \return false if fails to change collideWith (during execution).
+  bool setCollideWith(::dart::collision::CollisionGroupPtr collideWith);
 
-private: 
-  constexpr static double kMimicRatio = 0.333; 
-  constexpr static double kProximalVelocity = 0.01;
-  constexpr static double kDistalVelocity = kProximalVelocity*kMimicRatio;
+private:
+  constexpr static double kMimicRatio = 0.333;
+  constexpr static double kProximalSpeed = 1;
```

Comment:
I suspect we arbitrarily chose a value for testing, then. 😢

1. Our controller [calls the `trapezoidalMove()` function](https://github.com/personalrobotics/barrett_ros_control/blob/master/src/tasks/BHMoveHandTask.cpp#L78) to move the hand:
2. That function [sets the `E` Puck property to the desired position, then the `MODE` property to `MODE_TRAPEZOIDAL`](https://github.com/BarrettTechnology/libbarrett/blob/8e97d0f6d3b36fc14c5520d583691146627a97ca/src/products/hand.cpp#L156) to execute the action.
3. Here is [a snippet of the description](http://support.barrett.com/wiki/Hand/282/ControlModes) of the behavior of the hand in `MODE_TRAPEZOIDAL` (the full description goes into much more detail):

    > [...] The BarrettHand will generate its own trapezoidal velocity profile to accelerate from its present position (using ACCEL) until it reaches max velocity (MV), then it will decelerate to reach the target endpoint position at zero velocity. [...]

So the correct way to get this value is to query the `MV` property on the four pucks in the BarrettHand, then use the transmission ratio to convert the motor velocity to a joint velocity. We should do this at some point (maybe file an issue?), but it's not urgent to get this right.

In the meantime, we can estimate a value by timing how long it takes the hand to close. This is all an approximation, anyway, since we are assuming the finger immediately accelerates to its maximum velocity.

From watching a YouTube video of HERB, I estimate around 1 s. Since the joint range is 140 degrees, this is around 2.4 rad/s. 🔬